src/light-clients/ethereum.md

```diff
@@ -0,0 +1,143 @@
+
+# Ethereum Light Client
+
+This chapter explores the architecture and operation of the Ethereum light client, along with the cryptographic foundations that ensure the validity of its proofs. We'll begin with a high-level overview of the protocol before diving into the technical details of each component.
+
+## Protocol Overview
+
+A light client exposes several public functions that can be invoked by external parties. Each function call modifies the client's internal state `S`, storing critical data that informs subsequent operations. The light client lifecycle can be divided into three distinct phases:
+
+1. **Initialization**: Occurs exactly once when the client is created
+1. **Updating**: Called repeatedly to verify each new block
+1. **Freezing**: Invoked once if malicious behavior is detected
+
+Initialization typically happens during contract deployment for smart contract-based light clients. During this phase, we provide the client with its initial trusted state, which can be either:
+
+- The genesis block of the chain
+- A recent, well-established checkpoint block
+
+This initial block is known as the "trusted height" and serves as the foundation for all subsequent verifications. Since this block is assumed to be correct without cryptographic verification within the light client itself, its selection is critical. In production environments, a governance proposal or similar consensus mechanism often validates this block before it's passed to the light client.
+
+Once initialized, the light client can begin verifying new blocks. The update function accepts a block header and associated cryptographic proofs, then:
+
+1. Verifies the header's cryptographic integrity
+1. Validates the consensus signatures from the validator set
+1. Updates the client's internal state to reflect the new "latest verified block"
+
+Updates can happen in sequence (verifying each block) or can skip intermediate blocks using more complex proof mechanisms. The efficiency of this process is what makes light clients practical for cross-chain communication.
+
+If the light client detects conflicting information or invalid proofs that suggest an attack attempt, it can enter a "frozen" state. This is a safety mechanism that prevents the client from processing potentially fraudulent updates. Recovery from a frozen state typically requires governance intervention.
+
+Since initialization is rather trivial, we will not dive deeper into it.
+
+### Updating
+
+Since Ethereum is finalized by the beaconchain, our Ethereum light client accepts beacon block data as update input. A [beacon block](https://eth2book.info/capella/part3/containers/blocks/#beacon-blocks) rougly has this structure:
```

Comment:
This is the full beacon block data, we rather are interested in the `LightClientUpdate` data. This also includes the execution header not only the beacon block. It also contains the beacon header, not the full data.
https://github.com/unionlabs/union/blob/main/lib/beacon-api-types/src/deneb/light_client_header.rs#L12